ysh/val_ops.py

```diff
@@ -120,6 +120,11 @@ def Stringify(val, blame_loc, prefix=''):
             val = cast(value.Eggex, UP_val)
             s = regex_translate.AsPosixEre(val)  # lazily converts to ERE
 
+        elif case(value_e.List):
+            raise error.TypeErrVerbose(
+                "%got a List, which can't be Stringified! Use @ instead of $ or join()" % prefix,
```

Comment:
Typo -- should be %s and not %g

I would say

"Maybe use @ to splice, instead of $"

The problem is that this error can actually appear in many places

We have noticed in the past that overly specific error messages can become misleading -- i.e. trying to be helpful doesn't always work:

https://oilshell.zulipchat.com/#narrow/stream/384942-language-design/topic/Problem.20with.20our.20static.20checks

---

It would also help to add another function to `test/ysh-runtime-errors.sh`, with all the examples that are intended to be changed by this commit

Then we can run it and see how they look.

Reply:
Is there a possibility to check the AST if we're at a `$` ?
The "dream" scenario IMO would be something like:
```
ysh ysh-0.19.0$ write $[:|a b c|]                             write $[:|a b c|]
        ^~ Use @ instead of $ to slice
[ interactive ]:1: fatal: expected Null, Bool, Int, Float, Eggex, got List. Can't stringify.
```

Reply:
It's possible in theory ... we have all the info necessary

But I don't think we have any examples of it.  We would have to enumerate all the cases and make sure that the error makes sense.

I think this actually DOES exist in `test/ysh-runtime-errors.sh`, but I suppose it's not organized well enough to actually find them!  I guess we could grep the output and find them.

Whenever I write error messages, I put examples in that file, and eyeball them to see if they make sense.


---

As mentioned it's easy to construct an error for a specific situations, and then find it doesn't apply to other situations that hit the same code path

I think it's OK to say "Maybe" or "Hint" in those cases